internal/database/metadata/postgres/db.go

```diff
@@ -0,0 +1,48 @@
+package postgres
+
+import (
+	"context"
+
+	"github.com/oom-ai/oomstore/internal/database/metadata"
+)
+
+func (db *DB) CreateEntity(ctx context.Context, opt metadata.CreateEntityOpt) (int16, error) {
+	return createEntity(ctx, db, opt)
+}
+
+func (db *DB) UpdateEntity(ctx context.Context, opt metadata.UpdateEntityOpt) error {
+	return updateEntity(ctx, db, opt)
+}
+
+func (db *DB) CreateFeatureGroup(ctx context.Context, opt metadata.CreateFeatureGroupOpt) (int16, error) {
+	return createFeatureGroup(ctx, db, opt)
+}
+
+func (db *DB) UpdateFeatureGroup(ctx context.Context, opt metadata.UpdateFeatureGroupOpt) error {
+	return updateFeatureGroup(ctx, db, opt)
+}
+
+func (db *DB) CreateFeature(ctx context.Context, opt metadata.CreateFeatureOpt) (int16, error) {
+	return createFeature(ctx, db, opt)
+}
+
+func (db *DB) UpdateFeature(ctx context.Context, opt metadata.UpdateFeatureOpt) error {
+	return updateFeature(ctx, db, opt)
+}
+
+func (db *DB) CreateRevision(ctx context.Context, opt metadata.CreateRevisionOpt) (int32, string, error) {
+	var (
+		revisionId int32
+		dataTable  string
+		err        error
+	)
+	err = db.WithTransaction(ctx, func(c context.Context, s metadata.Store) error {
```

Comment:
Because the implementation of CreateRevision requires a transaction

Reply:
I don't get it, it doesn't look different from other methods, and they don't require `WithTransaction` ...

Reply:
Because `CreateRevison` is implemented in two steps:
1. insert into fature_group_revison, get revision_id
2. update field data_table of feature_group_revision if `CreateRevisonOpt.DataTable` is null

so it does require transactions, which is different from other methods